indigo_app/views/works.py

```diff
@@ -50,7 +51,14 @@ class WorkViewBase(PlaceViewBase, AbstractAuthedIndigoView, SingleObjectMixin):
 
     def determine_place(self):
         if 'place' not in self.kwargs:
-            self.kwargs['place'] = self.kwargs['frbr_uri'].split('/', 2)[1]
+            try:
+                uri = FrbrUri.parse(self.kwargs['frbr_uri'])
+                self.kwargs['place'] = uri.place
+            except ValueError:
+                # not a valid URI; make some guesses
+                parts = self.kwargs['frbr_uri'].split('/')
```

Comment:
Because all our FRBR URIs now must start with /akn, can we not always just do the exception part (without parsing the whole FRBR URI)?

Reply:
We can for sure always just do the exception part and still be backwards compatible even, but I thought parsing the URI would be safer / take the guesswork out of it (the exception part was just added to get the tests passing). Are you saying it would be better not to parse it? Is it expensive?